modules/cluster-logging-collector-chunk.adoc

```diff
@@ -0,0 +1,96 @@
+
+Fluentd file buffering stores records in _chunks_. Chunks are stored in _buffers_.
+
+You can tune file buffering in your cluster by editing environment variables
+in the `fluentd` daemonset:
+
+* *BUFFER_SIZE_LIMIT*. This parameter determines the maximum size of each chunk 
+file before Fluentd creates a new chunk. The default is `8M`. 
+This parameter sets the Fluentd `buffer_chunk_limit` variable.
++
+A high BUFFER_SIZE_LIMIT can collect more records per chunk file. However, 
+bigger records need more time to be sent to the logstore. 
+
+* *FILE_BUFFER_LIMIT*. This parameter determines the file buffer size per output.
+The default is `256Mi`.
++
+The permanent volume size must be larger than`FILE_BUFFER_LIMIT` multiplied by the
+number of outputs.
++
+A high FILE_BUFFER_LIMIT could translate to a higher BUFFER_QUEUE_LIMIT based the number of outputs. 
+However, if the node's space is under pressure Fluentd can fail.
++
+By default, the `number_of_outputs` is `1` if all the logs are sent to a single resource, and is incremented by `1` for each 
+additional resource. You might have multiple outputs if you use the Log Forwarding API, the Fluentd protocols, or 
+syslog protocol to forward logs to external locations.
++
+For example, the value of `number_of_outputs` is:
+
+** `1` - if all logs are sent to a single Elasticsearch pod
+** `2` - if application logs are sent to an Elasticsearch pod and ops logs are sent to
+another Elasticsearch pod
+** `4` - if application logs are sent to an Elasticsearch pod, ops logs are sent to
+another Elasticsearch pod, and both of them are forwarded to other Fluentd instances
+
+* *BUFFER_QUEUE_LIMIT*. This parameter is the maximum number of buffer chunks allowed.
+The BUFFER_QUEUE_LIMIT parameters is not directly tunable. It is calculated by 
+{product-title} based on the number of lgging outputs, the chunk size and the filesystem space available. 
+The default is `32` chunks. To change the `BUFFER_QUEUE_LIMIT`, you must change the value of `FILE_BUFFER_LIMIT`.
+This parameter sets the Fluentd `queue_limit_length` parameter.
++
+The environment variable `BUFFER_QUEUE_LIMIT` is calculated as `(FILE_BUFFER_LIMIT / (number_of_outputs * BUFFER_SIZE_LIMIT))`.
++
+If the `BUFFER_QUEUE_LIMIT` variable has the default set of values:
+
+** `FILE_BUFFER_LIMIT = 256Mi`
+** `number_of_outputs = 1`
+** `BUFFER_SIZE_LIMIT = 8Mi`
+
+Fluentd uses the link:https://docs.fluentd.org/buffer/file[*file* buffer plug-in]
+to configure how the chunks are stored. You can see the location of the buffer file
+in the following files:
+
+----
+etc/fluent/configs.d/openshift/output-es-config.conf:
+  @type file <1>
+  path '/var/lib/fluentd/buffer-output-es-config' <2>
+
+etc/fluent/configs.d/openshift/output-es-ops-config.conf:
+  @type file <1>
+  path '/var/lib/fluentd/buffer-output-es-ops-config' <2>
+----
+<1> Specifies the *file* buffer plugin. Do not change this value. 
+<2> The path where buffer chunks are stored.
+
```

Comment:
@periklis Do we need the location of the buffer files? If so, is this the best place to find them?

Reply:
See above comment

Reply:
Outdated